osb-service/src/main/java/de/evoila/cf/cpi/bosh/MongoDBDeploymentManager.java

```diff
@@ -45,52 +56,55 @@ protected void replaceParameters(ServiceInstance serviceInstance, Manifest manif
+            UsernamePasswordCredential exporterCredential = credentialStore.createUser(serviceInstance, CredentialConstants.EXPORTER_CREDENTIALS, "exporter");
+            PasswordCredential replicaSetKey = credentialStore.createPassword(serviceInstance,"replicaSetKey", 40);
+
+            InstanceGroup instanceGroup =  manifest.getInstanceGroup("mongodb").get();
+
+            if(credentialStore instanceof DatabaseCredentialsClient){
+                JobV2 mongoJob = instanceGroup.getJob("mongodb").get();
+                JobV2 exportJob = instanceGroup.getJob("mongodb_exporter").get();
+                JobV2 backupJob = instanceGroup.getJob("backup-agent").get();
+                Map<String, Object> mongodb_properties = getProperty(mongoJob.getProperties(),"mongodb");
+                Map<String, Object> exporter_properties = getProperty(exportJob.getProperties(), "mongodb_exporter");
+                Map<String, Object> backup_properties = getProperty(backupJob.getProperties(), "backup_agent");
+
+
+                Map<String, Object> auth = getProperty(mongodb_properties, "auth");
+
+                Map<String, Object> replset = getProperty(auth,"replica-set");
+                replset.put("keyfile", replicaSetKey.getPassword());
+
+                List<HashMap<String, Object>> admins= (List<HashMap<String, Object>>) auth.get("admin_users");
+                if(admins == null){
+                    admins = new LinkedList<>();
+                    auth.put("admin_users",admins);
+                }
+                admins.clear();
+                admins.add(craeteUser(rootCredential.getUsername(),rootCredential.getPassword()));
+                admins.add(craeteUser(exporterCredential.getUsername(),exporterCredential.getPassword()));
+
+                List<HashMap<String, Object>> backup_users= (List<HashMap<String, Object>>) auth.get("backup_users");
+                if(backup_users == null){
+                    backup_users = new LinkedList<>();
+                    auth.put("backup_users",backup_users);
+                }
+                backup_users.clear();
+                backup_users.add(craeteUser(backupCredential.getUsername(),backupCredential.getPassword()));
+
+                exporter_properties.put("user", exporterCredential.getUsername());
+                exporter_properties.put("password", exporterCredential.getPassword());
+                exporter_properties.put("database", "admin");
+
+                backup_properties.put("user", backupCredential.getUsername());
+                backup_properties.put("password", backupCredential.getPassword());
             }
 
-            replset.put("name", properties.get(REPLICA_SET_NAME));
-            serviceInstance.getParameters().put("replicaSet", properties.get(REPLICA_SET_NAME));
 
-            if (properties.containsKey(DATA_DIR)) {
```

Comment:
Removed but not added again? On purpose?

Reply:
Not on purpose but there is no value in choosing a directory.